.github/workflows/pr.yaml

```diff
@@ -155,3 +157,53 @@ jobs:
 
       - name: Run Javascript tests
         run: dotrun test-js
+
+  publish_report:
+    name: publish-e2e-report
+    # if job is cancelled we should skip this
+    if: success() || needs.e2e.outputs.job_status == 'failure'
+    needs: [e2e]
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    env:
+      HTML_REPORT_URL_PATH: reports/${{ github.head_ref || github.ref_name }}/${{ github.run_id }}/${{ github.run_attempt }}
```

Comment:
So we keep multiple runs on the same run available, nice :+1:

Reply:
Maybe use the PR number instead of the branch name? Then we can trigger the deletion on PR merge instead of branch removal. Just an idea, maybe simpler to keep as it is.

Reply:
I don't see too much of a functional difference between tracking by PR number or branch name since this job is triggered with a pull request so it has to be an PR for a specific branch. The report delete action first checks if a folder with the branch name exists before deleting anything, so I don't think there is a risk there. Changing the implementation shouldn't be too bad but I do need to go through another round of testing which could take some time. If you think there is a definite benefit in doing this change then I can make it happen?

Reply:
I just realized: The source branch can be in a fork, and then the current workflow would miss the "delete branch" event. So I think we have to track PRs and clean up when they are closed.

Reply:
On the cleanup job, you can use 
```
on:
  pull_request:
    types: [ closed ]
```

Reply:
the actions are now tracking by PR number.

![Screenshot from 2023-12-13 17-09-53](https://github.com/canonical/lxd-ui/assets/151511568/94521090-bbf8-4bbf-ab51-7cb43c578868)